### ISSUE TITLE
possible to change from adam optimizer to L-BFGS?

### ISSUE THREAD

**Issue by nickyvdz-art** [czczup/URST#3]
possible to change from adam optimizer to L-BFGS?

not particularly excellent at any of this (code etc.) - so even just a simple, yes or no, will suffice, even as to prevent me from wasting hours attempting to cut and paste stuff that will never work? 

thanks again

**Comment by czczup** [comment]
> possible to change from adam optimizer to L-BFGS?
> 
> not particularly excellent at any of this (code etc.) - so even just a simple, yes or no, will suffice, even as to prevent me from wasting hours attempting to cut and paste stuff that will never work?
> 
> thanks again

Sorry, I have not tried L-BFGS, Adam seems to be a more common choice in style transfer.

**Comment by nickyvdz-art** [comment]
thanks again - if its in your interest, https://github.com/cysmith/neural-style-tf

when i use this via L-BFGS optimizer, I think the results produce a better blend of style / content image (i suppose perhaps personal preference?) 

again, thats not to say I don't think your work here is great - i just notice the results are similar to other adam optimizers, and they are quite distinctly different (again, to my eyes i suppose) from the results produced with the L-BFGS one

thanks again for replying so soon

**Comment by nickyvdz-art** [comment]
I suppose the real question is; is it possible to apply the URST model to the style transfer used in: https://github.com/cysmith/neural-style-tf

?

**Comment by nickyvdz-art** [comment]
![ours-patch1000-padding32](https://user-images.githubusercontent.com/68634496/113466930-5ad1dc80-948b-11eb-92c6-694a4716dd37.jpg)

![94ecfb90e4631319e58494f748501d6f81f68c37](https://user-images.githubusercontent.com/68634496/113466936-645b4480-948b-11eb-82f6-d28511c04d3d.jpg)

![b01](https://user-images.githubusercontent.com/68634496/113466947-8a80e480-948b-11eb-8daa-9ecc4827da82.jpg)
![ds](https://user-images.githubusercontent.com/68634496/113466950-8e146b80-948b-11eb-8322-504bb11e21e6.jpg)

**Comment by czczup** [comment]
Thanks for your question, https://github.com/cysmith/neural-style-tf seems to be an implementation of Gatys's paper "Image Style Transfer Using Convolutional Neural Networks". URST framework probably won't apply to this method because is an iteration-based method and there is no Instance Normalization (IN) layer. 

Maybe you can try something else, for example, `Huang2017AdaIN` or `Wang2020Collaborative`?  Our URST framework is developed on top of these existing IN-based methods, and the effect is determined by the selected method. I'm sorry if I can't help you.

**Comment by czczup** [comment]
I think `Johnson2016Perceptual` might be more suitable for you. If possible, you can train a new model (use your style image) for it.

**Comment by nickyvdz-art** [comment]
Please, no sorry - thanks for discussing this with me - i am not a coder, i can barely tinker stuff so it suits an artistic project I am aiming at - you are saving me a lot of time by your explanations / info

it seems like I simply have to get a far bigger GPU RAM capacity and that there is simply no way around it...

that said though - your model is fantastic, I have thrown very large pixel dimensions at it, it doesn't fail - where Wang2020 had its limits (though, also, very impressive) 

thank you so much for your time, wishing you well for the future

**Comment by nickyvdz-art** [comment]
yes, I am also still wanting to try the Johnson method - but I've not trained any models with it yet and just have to work out how to get there

thanks again for all your input

**Comment by czczup** [comment]
I can help you train a model. It will take some time. But the results may not be satisfactory.

**Comment by nickyvdz-art** [comment]
yeh, i think i have made 'models' in the past using the Wang2020 - (i.e. you train the model for hours, then after that you can apply it to any 'content' image pretty much instantly?) 

unlike the iterations process of cysmith-neural

i think im going to encounter the same problem

i think, most likely wrongly, the major 'qualitative' difference lies in the optimizer (i.e. Adam vs. L-BFGS)

it would be cool to modify the johnson method and see if it can swap from adam to l-bfgs instead

**Comment by nickyvdz-art** [comment]
but again, I am really grateful for the total 'package' of previous methods, your URST - its excellent, and helping me to understand things better

**Comment by nickyvdz-art** [comment]
![ours-patch1000-padding32-03](https://user-images.githubusercontent.com/68634496/113468619-ed29ae80-9492-11eb-905e-a182ffec56ef.jpg)

this is a higher res result from Li2018Learning + URST 

i think its excellent, but again, 'content' wise quite different from other optimizer
